quisp/modules/QRSA/RealTimeController/RealTimeController.cc

```diff
@@ -24,50 +23,13 @@ void RealTimeController::handleMessage(cMessage *msg) {}
 
 void RealTimeController::EmitPhoton(int qnic_index, int qubit_index, QNIC_type qnic_type, int pulse) {
   Enter_Method("EmitPhoton()");
-  cModule *qnode = getQNode();  // Get the parent QNode that runs this RC.
-
-  try {
-    // EV<<"EmitPhoton for qnic["<<qnic_index<<"] and qubit["<<qubit_index<<"]\n\n\n";
-    cModule *qubit = nullptr;
-    if (qnic_type >= QNIC_N) error("Only 3 qnic types are currently recognized....");  // avoid segfaults <3
-    qubit = qnode->getSubmodule(QNIC_names[qnic_type], qnic_index)->getSubmodule("statQubit", qubit_index);
-    StationaryQubit *q = check_and_cast<StationaryQubit *>(qubit);
-    q->emitPhoton(pulse);
-  } catch (std::exception &e) {
-    error("Some error occurred in RealTimeController. Here!. Maybe the qnic/statQubit couldnt be found. Have you changed the namings?");
-  }
-}
-
-cModule *RealTimeController::getQNode() {
-  // We know that Connection manager is not the QNode, so start from the parent.
-  cModule *currentModule = getParentModule();
-  try {
-    // Assumes the node in a network has a type QNode
-    cModuleType *QNodeType = cModuleType::get("modules.QNode");
-    while (currentModule->getModuleType() != QNodeType) {
```

Comment:
we don't need to write how to get components or modules like this. 
just call `provider`'s method